runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/TransformTranslator.java

```diff
@@ -121,6 +121,9 @@
      */
     long addOutput(PValue value);
 
+    /** Like {@link #addOutput(PValue)}, but allows to override the PValue's coder. */
+    long addOutput(PValue value, Coder<?> valueCoder);
```

Comment:
This is a frightening override. I'm not 100% sure I understand why the coder of the elements and the `FullWindowedValueCoders` won't line up.